### PR TITLE
driver shift table frontend

### PR DESCRIPTION
**Pull request by ru1nw**
In this PR, I added a frontend webpage for driver shift table.

### CODE REVIEW

**Comment by pconrad** [comment]
@ru1nw Ian, see my comments on the slack with suggestions about how to overcome the test failures...

https://ucsb-cs156-s23.slack.com/archives/C0515SM2Z5M/p1685583124776489

**Review by SophiaWeiler — CHANGES_REQUESTED** [review]
I think that this should have been broken down into multiple PRs, the first being the component for the table + tests + storybook (which is missing, you need to include the link to this in your PR) and then the second being the page with the link in Navbar. Can you update the description for the PR so that we can get a sense of what is being changed here?

File: frontend/src/main/components/Shift/ShiftTable.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+
+    // Stryker disable all : hard to test for query caching
+    const toggleAdminMutation = useBackendMutation(
+        cellToAxiosParamsToggleAdmin,
+        {},
+        ["/api/admin/users"]
+    );
+    // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/components/Shift/ShiftTable.js]: Please change to `// Stryker restore all`

See: https://ucsb-cs156-s23.slack.com/archives/C050779M19P/p1685726928710669

File: frontend/src/main/pages/ShiftPage.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            // Stryker disable next-line all : don't test internal caching of React Query
+            ["/api/admin/users"],
+            // Stryker disable next-line StringLiteral,ObjectLiteral : since "GET" is default, "" is an equivalent mutation
+            { method: "GET", url: "/api/admin/users" },
+            []
+        );
+
+    return (
+        <BasicLayout>
+            {/* <h2>Shift</h2> */}
```
Comment (pconrad) [inline frontend/src/main/pages/ShiftPage.js]: Can we remove this commented out line?

**Review by pconrad — CHANGES_REQUESTED** [review]
A few things to address

**Comment by ru1nw** [comment]
Closing this PR due to the large amount of files being changed. Will move to other branches and create smaller PRs. See #42, #50

**Pull request closed without merging**